### PR TITLE
Refactor x drive controller

### PR DESCRIPTION
**Pull request by toyat522**
X drive controller is now a "feedforward" velocity controller that takes into vehicle dynamics (will add documentation on how this works eventually, but I referenced Flippy's theses).

The previous PID controller is now part of controller server's waypoint following action server. This takes in position goal and outputs desired velocities.

The control mux doesn't send zero messages anymore as this messes with smoothing. Also probably not needed now that we have lora comms and thrust commander sends zero command when estopped.

The circular pid is moved to the pid_controller.py file.

Thrust commander takes in a Float64 not an Int64 to unify with sim.

### CODE REVIEW

File: all_seaing_driver/all_seaing_driver/rover_lora_controller.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+class RoverLoraController(Node):
+    def __init__(self):
+        super().__init__('rover_lora_controller')
+
+        # Setup serial connection
+        self.serial_port = serial.Serial('/dev/ttyUSB0', 57600, timeout=1)
+        time.sleep(2)  # Allow serial port to stabilize
+        
+        self.data_size = struct.calcsize('Bddd')
```
Comment (JanelleCai) [inline all_seaing_driver/all_seaing_driver/rover_lora_controller.py:18]: What is Bddd?

**Review by JanelleCai — COMMENTED** [review]
looks good to me

File: all_seaing_driver/all_seaing_driver/rover_lora_controller.py (hunk 1 above)
Comment (toyat522) [reply all_seaing_driver/all_seaing_driver/rover_lora_controller.py:18]: Byte double double double. This is Wells stuff tho and tbh should not be in my merge but ehhhh.

Comment (toyat522) [reply all_seaing_driver/all_seaing_driver/rover_lora_controller.py:18]: it was needed for the test on tuesday

**Pull request merged**